backend-test/Controllers/CourseControllerTests.hs

```diff
@@ -33,6 +34,14 @@ indexTestCases =
        "CSC101\nCSC102\nCSC103\nCSC104\nCSC105\n")
     ]
 
+-- | List of test cases as (input courses, course name, expected JSON output)
```

Comment:
In terms of code organization, keep each controller function's test code separate. That is, make sure everything to do with the `index` function is together, and everything to do with the `retrieveCourse` function is also together.